fedimintd/src/lib.rs

```diff
@@ -70,6 +70,17 @@ const SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(10);
 #[derive(Parser)]
 #[command(version)]
 #[command(
+    group(
+        ArgGroup::new("bitcoind_password_auth")
+           .args(["bitcoind_password", "bitcoind_url_password_file"])
+           .multiple(false)
+    ),
+    group(
+        ArgGroup::new("bitcoind_auth")
+            .args(["bitcoind_url"])
+            .requires("bitcoind_password_auth")
+            .requires_all(["bitcoind_username", "bitcoind_url"])
```

Comment:
[nit] I think the `bitcoind_url` here is redundant since it's already captured above in `.args(["bitcoind_url"])`, but I don't think that will cause any issues